From 38fb342a72e0d3631cb79f34de6a2815b301882c Mon Sep 17 00:00:00 2001
From: Yelena Krivosheev <yelena@marvell.com>
Date: Wed, 26 Apr 2017 14:26:09 +0300
Subject: [PATCH 0991/1345] net: mvneta: fix HW pool buffer size configuration

commit  a316199841fe0014d2226519e3f2cc0170aba18b from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

Fix MVNETA_PORT_POOL_BUFFER_SZ_REG register update procedure

Change-Id: Iffa89296ccc2fda72097f38d16b778e159bf87fe
Signed-off-by: Yelena Krivosheev <yelena@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/38939
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Dmitri Epshtein <dima@marvell.com>
Reviewed-by: Wilson Ding <dingwei@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/net/ethernet/marvell/mvneta.c |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/net/ethernet/marvell/mvneta.c b/drivers/net/ethernet/marvell/mvneta.c
index 2f356f5..26443bc 100644
--- a/drivers/net/ethernet/marvell/mvneta.c
+++ b/drivers/net/ethernet/marvell/mvneta.c
@@ -997,6 +997,7 @@ static inline void mvneta_bm_pool_bufsize_set(struct mvneta_port *pp,
 	}
 
 	val = mvreg_read(pp, MVNETA_PORT_POOL_BUFFER_SZ_REG(pool_id));
+	val &= ~MVNETA_PORT_POOL_BUFFER_SZ_MASK;
 	val |= buf_size & MVNETA_PORT_POOL_BUFFER_SZ_MASK;
 	mvreg_write(pp, MVNETA_PORT_POOL_BUFFER_SZ_REG(pool_id), val);
 }
@@ -1104,6 +1105,8 @@ static int mvneta_bm_port_init(struct platform_device *pdev,
 
 	mvneta_bm_pool_bufsize_set(pp, pp->pool_long->buf_size,
 				   pp->pool_long->id);
+	netdev_info(pp->dev, "create long pool %d for port %d, buffer size %d\n",
+		    pp->pool_long->id, pp->id, pp->pool_long->buf_size);
 
 	/* If short pool id is not defined, assume using single pool */
 	if (of_property_read_u32(dn, "bm,pool-short", &short_pool_id))
@@ -1124,6 +1127,8 @@ static int mvneta_bm_port_init(struct platform_device *pdev,
 		mvneta_bm_pool_bufsize_set(pp, pp->pool_short->buf_size,
 					   pp->pool_short->id);
 	}
+	netdev_info(pp->dev, "create short pool %d for port %d, buffer size %d\n",
+		    pp->pool_short->id, pp->id, pp->pool_short->buf_size);
 
 	return 0;
 }
-- 
1.7.9.5

